railties/lib/rails/generators/rails/app/templates/config/environments/development.rb.tt

```diff
@@ -63,11 +63,14 @@ Rails.application.configure do
   # Highlight code that triggered database queries in logs.
   config.active_record.verbose_query_logs = true
 
-  <%- end -%>
+  <%- if options[:database] == "mysql" || options[:database] == "trilogy" -%>
+  # Database adapter used for MySQL. Supports `:mysql2` and `:trilogy`.
+  config.active_record.mysql_adapter = <%= options[:database] == "trilogy" ? ":trilogy" : ":mysql2" %>
```

Comment:
I think we should only generate that if you created you app with `--database=mysql2`.

Reply:
There is no `--database=mysql2`. There is only `--database=mysql` and `--database=trilogy`.

https://github.com/rails/rails/blob/e1a0f6f667d7e3ffeb0ef9bd870e1d1e125e82d5/railties/lib/rails/generators/database.rb#L7

So the default when you specify your DBMS as "MySQL" is to use `mysql2`, and trilogy is opt in.

Should we change the supported list of generators? To include: `[ "mysql", "mysql2", "trilogy", ...`? Then `mysql` defaults to trilogy, and we can use the alias, and all others would explicitly load the adapters.

I guess what I'm trying to say (or rather @adrianna-chang-shopify is saying as we are pairing) is that there are a few places we still assume `mysql == mysql2` that are no longer correct, especially if we are changing the default now.

Reply:
Yeah, I was introducing it in https://github.com/rails/rails/pull/50110

IMO:

  - `--database=mysql2` => I want to use `mysql2`
  - `--database=trilogy` => I want to use `trilogy`
  - `--database=mysql` => I want to use MySQL server, I don't care about the adapter, Rails should chose the best one for me.

Reply:
Right. I see that now.

I'll pull those changes over. Thanks.